tests/test_vault.cairo

```diff
@@ -269,3 +268,82 @@ fn test_protect_position_user_address_is_zero() {
     suite.vault.protect_position(deposit_address, user, user_amount);
     stop_cheat_caller_address(suite.vault.contract_address);
 }
+
+#[test]
+fn test_get_vault_token(){
+    let user: ContractAddress = HYPOTHETICAL_OWNER_ADDR.try_into().unwrap();
+    let token: ContractAddress = deploy_erc20_mock();
+    let suite = setup_test_suite(user, token);
+    start_cheat_caller_address(suite.vault.contract_address, user);
+    assert(suite.vault.get_vault_token() == token, 'Vault token mismatch');
+    stop_cheat_caller_address(suite.vault.contract_address);
+}
+
+#[test]
+#[should_panic(expected: ('Vault token mismatch',))]
+fn test_get_vault_token_fail(){
+    let user: ContractAddress = HYPOTHETICAL_OWNER_ADDR.try_into().unwrap();
+    let suite = setup_test_suite(user, deploy_erc20_mock());
+    start_cheat_caller_address(suite.vault.contract_address, user);
+    assert(suite.vault.get_vault_token() == tokens::ETH.try_into().unwrap(), 'Vault token mismatch');
+    stop_cheat_caller_address(suite.vault.contract_address);
+}
+
+#[test]
+fn test_return_liquidity(){
+    let user: ContractAddress = HYPOTHETICAL_OWNER_ADDR.try_into().unwrap();
+    let user2: ContractAddress = MOCK_USER.try_into().unwrap();
+    let amount: u256 = 100;
+    let withdrawn_amount: u256 = 50;
+    let suite = setup_test_suite(user, deploy_erc20_mock());
+    setup_user(@suite, user, amount);
+
+    start_cheat_caller_address(suite.vault.contract_address, user);
+    suite.vault.store_liquidity(amount);
+    stop_cheat_caller_address(suite.vault.contract_address);
+
+    start_cheat_caller_address(suite.vault.contract_address, user);
+    suite.vault.return_liquidity(user2, withdrawn_amount);
+    stop_cheat_caller_address(suite.vault.contract_address);
+
+    let excpected_amount_user: felt252 = (amount-withdrawn_amount).try_into().unwrap();
+
+    assert_vault_amount(suite.vault.contract_address, user, excpected_amount_user);
+}
+
+#[test]
+#[should_panic(expected: ('Approved amount insufficient',))]
```

Comment:
There, you don't seem to have `Approved amount insufficient` error generated from some custom check, so now you are testing the `transfer_from` library functionality which is redundant.

Reply:
Okay, i remove it

Reply:
But wait, i saw this validation in `store_liquidity`. Should i do something similar? I mean this check:
```
assert(
   token_dispatcher.allowance(user, vault_contract) >= amount,
   'Approved amount insufficient'
);
```

Reply:
It's validation before `transfer_from`, as in my case

Reply:
Add an assert with a custom message that is more informational than the original, or remove this test